proposals/accepted/047-standard-library-reform.rst

```diff
@@ -0,0 +1,559 @@
+Reexporting modules from a less stable library (``base``) in more stable libraries is very error-prone.
+
+The generalization of these concerns is *rationalizing* dependencies, or rationalizing the division of labor between libraries.
+Once the purposes of libraries and the division of labor between them make more sense, it will be easier to maintain these libraries.
+It should be in fact easier than it was before to maintain them.
+
+New Goal: Split Base
+~~~~~~~~~~~~~~~~~~~~
+
+We should still split ``base``.
+This might sound surprising --- wasn't the point of making new libraries that we didn't need to worry about ``base`` so much?
+But it follows from the expanded "rationalize dependencies" goal.
+
+#. It will take a while for code to be migrated off ``base``, and until that process is complete ``base`` cannot serve as a "holding pen" for GHC's private implementation details.
+   Thus, until that process is complete, we would not have a solution to **Problem 2**.
+   Rather than waiting for ``base`` to stop being used, we can split it, and then GHC devs have (at least one) *proper* place for their unstable stuff, making a far more robust **Problem 2** solution while the migration away from ``base`` is still underway.
+
+#. Solving **Problem 4** requires that some of the code in ``base`` to day *not* be coupled with GHC and some of the code in ``base`` conversely *must* be coupled with GHC.
+   Thus solving **Problem 4** requires splitting ``base`` eventually anyways.
+
+#. ``base`` is treated specially in a few ways.
+   For example:
+
+   - It is the library that GHCi loads by default.
+
+   - GHC's compilation is directly aware of it in the form of various "wired-in" identifiers.
+
+   - Some modules of it are automatically trusted with Safe Haskell.
+
+   In the new multi-library world, different libraries will inherit these special features, and we cannot be sure what the ramification will be until we try.
+
+   It is best to "practice" this by splitting ``base`` as soon as possible.
+   That will reduce the risk of everything else by both exploring "known unknowns" and scouting ahead for "unknown unknowns".
+
+#. Ultimately, in the name of rationalizing dependencies and the library division of labor, ``base`` will never make sense in anything like its current form.
+   We should therefore demote it to being a mere reexporter of other libraries that do make sense.
+
+**Step 1B**: MVP Split ``base`` by making it all reexports
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The first steps of `GHC issue #20647`_ track what needs to be done here.
+The key first step is finishing `GHC MR !7898`_.
+This is crude: a ``ghc-base`` that ``base`` merely reexports in full is just as ugly as the original ``base``, but this is the quickest route to de-risking the entire project as described in item 2 of the previous section.
```

Comment:
Reducing risk is good. No need to apologize for it! This is a reasonable approach!